include/meta/meta.hpp

```diff
@@ -311,35 +417,47 @@ namespace meta
 
         namespace lazy
         {
+            /// \sa 'meta::bind_front'
+            /// \ingroup lazy_composition
             template <typename Fn, typename... Ts>
             using bind_front = defer<bind_front, Fn, Ts...>;
 
+            /// \sa 'meta::bind_back'
+            /// \ingroup lazy_composition
             template <typename Fn, typename... Ts>
             using bind_back = defer<bind_back, Fn, Ts...>;
         }
 
-        /// A metafunction that unpacks the types in the type list
-        /// \p List into the Metafunction Class \p F.
-        template <typename F, typename List>
-        struct apply_list_impl
+        ////////////////////////////////////////////////////////////////////////////
+        // apply_list
+        ///\cond
+        namespace detail
         {
-        };
+            /// A metafunction that unpacks the types in the type list
+            /// \p List into the Metafunction Class \p F.
+            template <typename F, typename List>
+            struct apply_list_
+            {
+            };
 
-        template <typename F, template <typename...> class T, typename... Ts>
-        struct apply_list_impl<F, T<Ts...>> : lazy::apply<F, Ts...>
-        {
-        };
+            template <typename F, template <typename...> class T, typename... Ts>
+            struct apply_list_<F, T<Ts...>> : lazy::apply<F, Ts...>
+            {
+            };
 
-        template <typename F, typename T, T... Is>
-        struct apply_list_impl<F, integer_sequence<T, Is...>>
-            : lazy::apply<F, std::integral_constant<T, Is>...>
-        {
-        };
+            template <typename F, typename T, T... Is>
+            struct apply_list_<F, integer_sequence<T, Is...>>
+                : lazy::apply<F, std::integral_constant<T, Is>...>
+            {
+            };
+        } // namespace detail
+        /// \endcond
 
         /// Applies the Metafunction Class \p C using the types in
         /// the type list \p List as arguments.
+        /// \ingroup invocation
         template <typename C, typename List>
-        using apply_list = eval<apply_list_impl<C, List>>;
+        using apply_list = eval<detail::apply_list_<C, List>>;
```

Comment:
No, I actually intend for `apply_list_impl` to be part of the pulic API, although the name needs work. End users should be able to specialize it to get `as_list`, `apply_list`, `curry`, `uncurry` and all the rest to work with their types. Notice how it's specialized for `integer_sequence`.